src/backend/distributed/executor/executor.c

```diff
@@ -1679,7 +1688,14 @@ StartPlacementExecutionOnSession(TaskPlacementExecution *placementExecution,
 	/* connection is going to be in use */
 	workerPool->idleConnectionCount--;
 
-	if (paramListInfo != NULL)
+
+	if (task->copyTask)
+	{
+		uint64 shardId = taskPlacement->shardId;
+		ExecuteCopyTask(connection, task->copyReceiver, shardId, task->copyData);
```

Comment:
I think doing it prevents any parallelism across task executions. Shouldn't we call different pieces of the function on different parts of `TransactionStateMachine`?

For example, `ConstructCopyStatement` might be good in `StartPlacementExecutionOnSession`. Later, on `REMOTE_TRANS_SENT_COMMAND`, do `SendCopyDataToPlacement` and on `REMOTE_TRANS_CLEARING_RESULTS` do `PutRemoteCopyEnd` etc?

Because, we're currently only utilizing 1 connection per node. If you set the debug level to `DEBUG4`, you'd see something like:
```
DEBUG:  Total number of commands sent over the session 253: 10
DEBUG:  Total number of commands sent over the session 254: 0
DEBUG:  Total number of commands sent over the session 255: 0
DEBUG:  Total number of commands sent over the session 256: 0
DEBUG:  Total number of commands sent over the session 257: 11
DEBUG:  Total number of commands sent over the session 258: 0
DEBUG:  Total number of commands sent over the session 259: 0
DEBUG:  Total number of commands sent over the session 260: 0
DEBUG:  Total number of commands sent over the session 261: 11
DEBUG:  Total number of commands sent over the session 262: 0
DEBUG:  Total number of commands sent over the session 263: 0
DEBUG:  Total number of commands sent over the session 264: 0
```

Reply:
Thanks. I agree, I will address this.